### PR TITLE
ORC-1694: Upgrade gson to 2.9.0 for Benchmarks Hive

### PR DESCRIPTION
**Pull request by cxzl25**
### What changes were proposed in this pull request?
This PR aims to upgrade gson to 2.9.0 for Benchmarks Hive.

### Why are the changes needed?
ORC-1676 upgraded to Hive4, Hive4 uses gson 2.9.0.

HIVE-26322: Upgrade gson to 2.9.0

https://issues.apache.org/jira/browse/HIVE-26322

### How was this patch tested?
GA

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Review by dongjoon-hyun — COMMENTED** [review]
The `bench` module also has Apache Spark. Is this compatible?

https://mvnrepository.com/artifact/org.apache.spark/spark-hive_2.12/3.5.1

BTW, I checked Apache Spark master branch too. Apache Spark 4.0.0 will use old gson too still.

```
[INFO] org.apache.spark:spark-hive_2.13:jar:4.0.0-SNAPSHOT
[INFO] +- org.apache.hive:hive-exec:jar:core:2.3.9:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.2.4:compile
```

**Comment by cxzl25** [comment]
> I checked Apache Spark master branch too. Apache Spark 4.0.0 will use old gson too still

Thanks to dongjoon for mentioning this.

> Is this compatible?

I noticed that Hive was upgraded from gson 2.2.4 to 2.9.0, only the version was modified, UT did not fail, I will test it in ORC benchmark.

[HIVE-26078](https://issues.apache.org/jira/browse/HIVE-26078): Upgrade gson to 2.8.9

[HIVE-26322](https://issues.apache.org/jira/browse/HIVE-26322): Upgrade gson to 2.9.0

**Comment by dongjoon-hyun** [comment]
Thanks. If we can run the Hive benchmark without any issues, yes, we can upgrade this. :)

**Comment by cxzl25** [comment]
> If we can run all benchmarks without any issues

I tested all the commands against the benchmark's README and found a few problems, though not related to the gson version.

Except ORC-1696 has been merged, I can retest after ORC-1699 ORC-1700 is merged.

**Comment by dongjoon-hyun** [comment]
Since this is blocked by the others, I'll put it to `Draft` status to avoid any accidental merging.
> I can retest after [ORC-1697](https://issues.apache.org/jira/browse/ORC-1697) [ORC-1700](https://issues.apache.org/jira/browse/ORC-1700) is merged.

**Comment by dongjoon-hyun** [comment]
Is this ready back, @cxzl25 ?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

I guess you already tested in bench module for your testing.

Feel free to merge this to `main` and `branch-2.0`, @cxzl25 .

**Comment by cxzl25** [comment]
> Is this ready back

Yes

> I guess you already tested in bench module for your testing.

Yes, because the PR mentioned before have been merged, I used the benchmark's README to execute it from beginning to end and found no exceptions.

**Pull request closed without merging**
